frontend-web/src/shared/persistence.ts

```diff
@@ -0,0 +1,18 @@
+const keys = {
+  username: 'username'
+} as const
+
+type ActualKeys = typeof keys[keyof typeof keys];
+const inMemoryCache: { [key in ActualKeys]?: string } = {};
```

Comment:
Nice :) (but might be a little bit over engineered since localstorage is pretty fast 😜)